### ISSUE TITLE
RStudio Server Preview 1.3.1045 fails to start

### ISSUE THREAD

**Issue by andresrcs** [rstudio/rstudio#7221]
### System details

    RStudio Edition : Server
    RStudio Version : 1.3.1045
    OS Version      : Ubuntu 18.04.4 LTS
    R Version       : 4.0.2

### Steps to reproduce the problem
* Install the preview version
```
wget https://s3.amazonaws.com/rstudio-ide-build/server/bionic/amd64/rstudio-server-1.3.1045-amd64.deb
sudo gdebi rstudio-server-1.3.1045-amd64.deb
```
### Describe the problem in detail
After upgrading no error message is returned on the terminal but it's not possible to open an RStudio session, the browser just keeps loading indefinitely. 

This is what I get after running `sudo rstudio-server verify-installation`

```
sudo rstudio-server verify-installation
25 Jun 2020 01:25:51 [rsession-rstudio-server] ERROR system error 13 (Permission                                                   denied) [path: /home/rstudio-server/.config/rstudio, target-dir: ]; OCCURRED AT                                                   rstudio::core::Error rstudio::core::FilePath::createDirectory(const string&) co                                                  nst src/cpp/shared_core/FilePath.cpp:648; LOGGED FROM: rstudio::core::Error rstu                                                  dio::session::prefs::initializePrefs() src/cpp/session/prefs/UserPrefs.cpp:142
25 Jun 2020 01:25:51 [rsession-rstudio-server] ERROR system error 13 (Permiso de                                                  negado) [path: /home/rstudio-server/.config/rstudio/dictionaries/custom, target-                                                  dir: ]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::createDirector                                                  y(const string&) const src/cpp/shared_core/FilePath.cpp:648; LOGGED FROM: rstudi                                                  o::core::Error rstudio::core::spelling::HunspellDictionaryManager::availableLang                                                  uages(std::vector<rstudio::core::spelling::HunspellDictionary>*) const src/cpp/c                                                  ore/spelling/HunspellDictionaryManager.cpp:158
25 Jun 2020 01:25:51 [rsession-rstudio-server] ERROR system error 13 (Permiso de                                                  negado) [path: /home/rstudio-server/.config/rstudio/dictionaries/custom, target-                                                  dir: ]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::createDirector                                                  y(const string&) const src/cpp/shared_core/FilePath.cpp:648; LOGGED FROM: std::v                                                  ector<std::__cxx11::basic_string<char> > rstudio::core::spelling::HunspellCustom                                                  Dictionaries::dictionaries() const src/cpp/core/spelling/HunspellCustomDictionar                                                  ies.cpp:33
```
If I downgrade to version 1.3.1039 everything goes back to normal

### Describe the behavior you expected

- [x] I have read the guide to submitting good bug reports at https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports . 
- [x] I have installed the latest version of RStudio and confirmed that the issue still persists.
- [ ] If I am reporting a RStudio crash, I have included a diagnostics report. https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by ronblum** [comment]
@andresrcs Thanks! I'm marking this as a 1.4 bug.

**Comment by ronblum** [comment]
@andresrcs Oops—I meant 1.3 (Water Lily).

**Comment by rich-rstudio** [comment]
I was able to reproduce this on 1.3.1045, although I was still able to launch a new session after quitting the existing one, but I do see the same in errors

**Comment by kfeinauer** [comment]
Similar to https://github.com/rstudio/rstudio-pro/issues/1718 which was fixed.

**Comment by kfeinauer** [comment]
@ricardofandrade has offered to take a look since it is a similar issue to the one he recently fixed

**Comment by kevinushey** [comment]
It seems like this is worth fixing for v1.3-water-lily, no? Unless the circumstances leading to this failure-to-launch are very esoteric.

**Comment by kfeinauer** [comment]
The only errors I actually see are `verify-installation` errors - the session seems to start up fine as per normal. The actual errors emitted in `verify-installation` are harmless user directory creation issues - see code fix for more details.

**Comment by kevinushey** [comment]
The original report says, though:

> After upgrading no error message is returned on the terminal but it's not possible to open an RStudio session, the browser just keeps loading indefinitely.

Are there two independent issues here? Is the `verify-installation` issue a red-herring for this failure to load?

**Comment by kfeinauer** [comment]
The `verify-installation` errors we see (and that I fixed in the attached PR) are definitely not related to any session load failure issues if they are occurring. Myself and @rich-rstudio where unable to reproduce the inability to start new sessions.

**Comment by andresrcs** [comment]
Can you guide me on how to provide additional information relevant to the inability to start a session?

**Comment by kfeinauer** [comment]
@andresrcs 

After starting RStudio Server (via `sudo rstudio-server restart`), do you see any error logs in syslog (`/var/log/syslog`)? After signing in, while you see the session spinner which never completes, is anything logged then?

If we have no error logs to go off of, can you try resetting your user state folder by deleting `$HOME/.config/rstudio`? You can also move the folder if you have any important state that you feel you may need access to in the future, such as any unsaved document changes.

**Comment by andresrcs** [comment]
>  do you see any error logs in syslog (/var/log/syslog)?

No, just this
```
 un 25 21:26:47 ip-172-31-10-23 systemd[1]: Started Session 100 of user ubuntu.
Jun 25 21:26:57 ip-172-31-10-23 systemd[1]: Stopping RStudio Server...
Jun 25 21:26:57 ip-172-31-10-23 systemd[1]: Stopped RStudio Server.
Jun 25 21:26:57 ip-172-31-10-23 systemd[1]: Starting RStudio Server...
Jun 25 21:26:57 ip-172-31-10-23 systemd[1]: Started RStudio Server.
````

After some waiting (couple minutes) the session finally started, I haven't done anything else since yesterday so I suppose this is going to remain a mystery for me, maybe just a connectivity issue with AWS that I misrelated with RStudio.

**Comment by kfeinauer** [comment]
Glad to hear your session started. If you have any other issues, please do let us know!

**Comment by R-Hannibal** [comment]
Verified Fixed on: RStudio Version - 1.3.1073-1, R Version - 4.0.2, RHEL 8.2 - Accessed from Windows 10 x64, Chrome 

`rstudio-server verify-installation` is running/completing successfully on the latest preview and current stable build (1.3.1056). No errors are thrown.

*Updating milestone to waterlily as the fix was originally merged to 1.3 there.
